test/Microsoft.Azure.Devices.Edge.Test.Common/EdgeModule.cs

```diff
@@ -166,7 +166,9 @@ async Task WaitForStatusAsync()
                             if (v.Path.EndsWith("settings.createOptions"))
                             {
                                 // normalize stringized JSON inside "createOptions"
-                                v.Value = JObject.Parse((string)v.Value).ToString(Formatting.None);
+                                // But ignore the value if it starts with 'net.azure-devices.edge'
+                                string value = JObject.Parse((string)v.Value).ToString(Formatting.None);
+                                v.Value = value.Contains("net.azure-devices.edge") ? "{}" : value;
```

Comment:
Yeah this should work but it is pretty coarse. 😉 

Instead of filtering certain values _out of_ the _actual_ twin json, what if we insert it _into_ the _expected_ json? I mean, we know exactly what values we expect to find there.

In `EdgeConfiguration.cs`, in the method `Create`, you could replace the following line:

```csharp
            var reported = new Dictionary<string, object>
            {
                ["systemModules"] = desired
                    .Value<JObject>("systemModules")
                    .Children<JProperty>()
                    .ToDictionary(
                        p => p.Name,
-                        p => CreateExpectedModuleConfig((JObject)p.Value))
+                        p => p.Name == ModuleName.EdgeAgent
+                            ? CreateExpectedAgentModuleConfig((JObject)p.Value)
+                            : CreateExpectedModuleConfig((JObject)p.Value))
            };
```

...then create the new method `CreateExpectedAgentModuleConfig` that does this (untested):

```csharp
        static object CreateExpectedAgentModuleConfig(JObject source)
        {
            JToken createOptions = source.SelectToken("settings.createOptions") ?? new JObject();
            JToken env = source.SelectToken("env") ?? new JObject();

            string createOptionsLabel = JsonConvert.SerializeObject(createOptions);

            createOptions.Value<JObject>().TryAdd("Labels", new JObject());
            JObject labels = createOptions.Value<JObject>("Labels");
            labels.TryAdd("net.azure-devices.edge.owner", new JValue("Microsoft.Azure.Devices.Edge.Agent"));
            labels.TryAdd("net.azure-devices.edge.create-options", new JValue(createOptionsLabel));
            labels.TryAdd("net.azure-devices.edge.env", env);

            return CreateExpectedModuleConfig(source);
        }
```

Reply:
Thanks for the suggestion! Had to alter a bit to get it to work, but it works